### ISSUE TITLE
How can I get the 'gt_mini_train. pickle' to run adaptor.py and then generate 'mini-train' under 'trj_data' folder?

### ISSUE THREAD

**Issue by Jctrp** [alfredgu001324/MapBEVPrediction#2]
I hope to use the Mini nuScenes dataset to train the HiVT model for maptr_bev version.  How can I achieve this?  Additionally, when using a pre-trained checkpoint file named maptr_bev.ckpt to test the HiVT model, an error occurred due to mismatched model weights.

**Comment by alfredgu001324** [comment]
Thank you for your interest in our work!!

You can follow [the dataset prep step](https://github.com/alfredgu001324/MapBEVPrediction/blob/main/docs/prepare_dataset.md) and see also how to merge that with [trajectory data](https://github.com/alfredgu001324/MapBEVPrediction/blob/main/docs/adaptor.md) to produce the mini dataset. 

For the second point, thanks for letting me know!! I think I might have changed some class names when tidying up my code. Really appreciate for pointing this out. I will try to fix this as soon as possible

**Comment by ericaweng** [comment]
Did you actually address OP's question? I notice you provide gt_mini_val.pickle [here ](https://drive.google.com/drive/folders/17kqpilI4dP6ZY7XFnWArPbMSlpDq8ErM?usp=drive_link) but not gt_mini_train.pickle. However, we need gt_mini_train.pickle in order to generate the preprocessed adaptor files for training the prediction models on the mini dataset. Could you provide the code for generating  gt_mini_train.pickle? In particular, I'm having difficulty figuring out which map elements to use for generating the 'boundary' class.

**Comment by alfredgu001324** [comment]
Hi, thanks for reaching out. Sorry I have been a bit lazy on this and dragged onto my new projects... I actually did not generate `gt_mini_train.pickle` at all for several reasons:
1. mini train in nuscenes is too small that even if you train it for many epochs, it still cannot give a reasonable model. 
2. I initially generated the mini pickle files for debugging purposes, I did not actually train and eval on the mini. 

In terms of how to generate it, you can take a look at [this file](https://github.com/hustvl/MapTR/blob/main/tools/maptr/vis_pred.py), specifically [this part of the code](https://github.com/hustvl/MapTR/blob/1b435fd9f0db9a14bb2a9baafb565200cc7028a2/tools/maptr/vis_pred.py#L273) where they extract the gt coordinates. What I did is that I just run this file and save the gt coordinates into the pickle files that I uploaded to google drive. This is just for easier use of the repo. 

For the last question, the data structure looks as follows:
```
(Pdb) data['b4ff30109dd14c89b24789dc5713cf8c'].keys()
dict_keys(['gt_lines', 'gt_labels'])
(Pdb) data['b4ff30109dd14c89b24789dc5713cf8c']['gt_labels']
array([0, 0, 2, 2, 2, 2, 2])
(Pdb) data['b4ff30109dd14c89b24789dc5713cf8c']['gt_lines'].shape
(7, 20, 2)
```

So basically the first key is the sample_token/sample_idx of nuscenes, which corresponds to one single scenario/one snapshot. `gt_labels` can be 0, 1, 2 indicating the type of classes [boundary, crosswalks, divider] (I do not remember the exact order but you can take a look the the visualization code but simply trial and error). `gt_lines` are the coordinates of the map elements.

Let me know if you have further questions!
